src/stochprog.jl

```diff
@@ -0,0 +1,105 @@
+using LightGraphs
+
+export AbstractStochasticProgram, stochasticprogram
+export isleaf, probability, cutgenerator, numberofpaths
+export setprobability!, add_scenario_state!, add_scenario_transition!
+
+"""
+    AbstractStochasticProgram
+
+Stochastic program instance
+"""
+abstract type AbstractStochasticProgram <: AbstractGraph{Int} end
+
+"""
+    stochasticprogram(args...)
+
+Creates a stochastic program from the arguments
+"""
+function stochasticprogram end
+
+"""
+    getmaster(sp::AbstractStochasticProgram)
+
+Returns the master node of stochastic problem `sp`.
+"""
+function getmaster end
+
+"""
+    probability(sp::AbstractStochasticProgram, edge)
+
+Returns the probability to take the edge `edge` in the stochastic problem `sp`.
+"""
+function probability end
+
+"""
+    setchildx!(sp::AbstractStochasticProgram, node, child, sol)
+
+Sets the parent solution of `child` as `sol`, the solution obtained at `node`.
+"""
+function setchildx! end
+
+"""
+    solve!(sp::AbstractStochasticProgram, node)
+
+Solves the program at node `node` in `sp` and returns the solution.
+"""
+function solve! end
+
+"""
+    getobjectivebound(sp::AbstractStochasticProgram, node)
+
+Gets the current bound to the objective of `node`.
+"""
+function getobjectivebound end
+
+"""
+    setθbound!(sp::AbstractStochasticProgram, node, child, θlb)
+
+Sets the bounds to the objective of the child `child` of `node` to `θlb`.
+"""
+function setθbound! end
```

Comment:
Maybe we should explicit `theta` as `BellmanFunction`, but I am not sure. What do you think?

Reply:
Yes, bellman function seems better.